flink-test-utils-parent/flink-connector-test-utils/src/main/java/org/apache/flink/connector/testframe/testsuites/SourceTestSuiteBase.java

```diff
@@ -462,11 +477,18 @@ public void testSourceMetrics(
                             // skip failed assert try
                             return false;
                         }
-                    });
+                    },
+                    METRICS_TIMEOUT,
+                    DELAY_BETWEEN_CHECKS,
+                    String.format(
+                            "Timeout while comparing source metrics with %d expected value",
+                            getTestDataSize(testRecordCollections)));
```

Comment:
We're trying to get rid of timeouts in tests.

Reply:
And by the way, this change you're commenting belongs to [the timeout pr](https://github.com/apache/flink/pull/21035) on which this PR is based. We should discuss it there.